common/runtime/model.go

```diff
@@ -158,63 +158,178 @@ type OIDCConfigDTO struct {
+		} else {
+			o.validateIssuerURLForIndex(oidc.IssuerURL, i, errs)
+		}
+		o.validateSigningAlgsForIndex(oidc.SigningAlgs, i, errs)
+		o.validateRequiredClaimsForIndex(oidc.RequiredClaims, i, errs)
+	}
+}
+
+func (o *OIDCsDTO) validateIssuerURL(issuerURL string, errs *[]string) {
+	issuer, err := url.Parse(issuerURL)
+	if err != nil || (issuer != nil && len(issuer.Host) == 0) {
+		*errs = append(*errs, "issuerURL must be a valid URL")
+	}
+	if issuer != nil {
+		if issuer.Fragment != "" {
+			*errs = append(*errs, "issuerURL must not contain a fragment")
 		}
-		if issuer != nil && issuer.Fragment != "" {
-			errs = append(errs, "issuerURL must not contain a fragment")
+		if issuer.User != nil {
+			*errs = append(*errs, "issuerURL must not contain a username or password")
 		}
-		if issuer != nil && issuer.User != nil {
-			errs = append(errs, "issuerURL must not contain a username or password")
+		if len(issuer.RawQuery) > 0 {
+			*errs = append(*errs, "issuerURL must not contain a query")
 		}
-		if issuer != nil && len(issuer.RawQuery) > 0 {
-			errs = append(errs, "issuerURL must not contain a query")
+		if issuer.Scheme != "https" {
+			*errs = append(*errs, "issuerURL must have https scheme")
 		}
-		if issuer != nil && issuer.Scheme != "https" {
-			errs = append(errs, "issuerURL must have https scheme")
+	}
+}
+
+func (o *OIDCsDTO) validateIssuerURLForIndex(issuerURL string, index int, errs *[]string) {
+	issuer, err := url.Parse(issuerURL)
+	if err != nil || (issuer != nil && len(issuer.Host) == 0) {
+		*errs = append(*errs, fmt.Sprintf("issuerURL must be a valid URL for OIDC at index %d", index))
+	}
+	if issuer != nil {
+		if issuer.Fragment != "" {
+			*errs = append(*errs, fmt.Sprintf("issuerURL must not contain a fragment for OIDC at index %d", index))
+		}
+		if issuer.User != nil {
+			*errs = append(*errs, fmt.Sprintf("issuerURL must not contain a username or password for OIDC at index %d", index))
+		}
+		if len(issuer.RawQuery) > 0 {
+			*errs = append(*errs, fmt.Sprintf("issuerURL must not contain a query for OIDC at index %d", index))
+		}
+		if issuer.Scheme != "https" {
+			*errs = append(*errs, fmt.Sprintf("issuerURL must have https scheme for OIDC at index %d", index))
 		}
 	}
-	if len(o.SigningAlgs) != 0 {
+}
+
+func (o *OIDCsDTO) validateSigningAlgs(signingAlgs []string, errs *[]string) {
+	if len(signingAlgs) != 0 {
 		validSigningAlgs := o.validSigningAlgsSet()
-		for _, providedAlg := range o.SigningAlgs {
+		for _, providedAlg := range signingAlgs {
 			if !validSigningAlgs[providedAlg] {
-				errs = append(errs, "signingAlgs must contain valid signing algorithm(s)")
+				*errs = append(*errs, "signingAlgs must contain valid signing algorithm(s)")
 				break
 			}
 		}
 	}
+}
 
-	if len(errs) > 0 {
-		err := fmt.Errorf(strings.Join(errs, ", "))
-		return err
+func (o *OIDCsDTO) validateSigningAlgsForIndex(signingAlgs []string, index int, errs *[]string) {
```

Comment:
Please consider not passing index.
Just returning error and decorating this error with index on the caller level.